pkg/controller/watchnamespaces.go

```diff
@@ -30,10 +34,24 @@ func newWatchNamespacesCache() *watchNamespacesCache {
 	}
 }
 
+// getFormattedNamespaces returns a list of namespaces filtering through given list
+// and formatting them as namespace structs
+func getFormattedNamespaces(list corev1.NamespaceList, ignorePattern *regexp.Regexp) (fns []namespace) {
+	for _, ns := range list.Items {
+		name := ns.GetName()
+		if ignorePattern == nil || !ignorePattern.Match([]byte(name)) {
+			fns = append(fns, namespace{uid: string(ns.GetUID()), name: name})
```

Comment:
Not much related to this update, but perhaps we could have the `uid` attribute as `types.UID` instead of string.

Reply:
i have no strong opinion here. I think the uid is mainly used as a value for a prometheus label. So `string` should be good enough. 🧑‍💻

Reply:
I added a [note](https://github.com/app-sre/deployment-validation-operator/pull/221/files#diff-442af8c760ab7d47870e117c7b97eead9959b6f3bea8961f180efb7bb32e2c80R13) for future change. I didn't want to change any signature on this PR.